model/textparse/openmetricsparse.go

```diff
@@ -94,13 +95,19 @@ type OpenMetricsParser struct {
 	exemplarVal   float64
 	exemplarTs    int64
 	hasExemplarTs bool
+
+	// lines ending with _created are skipped by default as they are
+	// parsed by the CreatedTimestamp method. The skipCT flag is set to
+	// false when the CreatedTimestamp method is called.
+	skipCT bool
```

Comment:
To be addressed?

Reply:
About this, essentially whenever we call an OM parser we want skipCT to always be true. The only case the parser will not skip CT is when it is a deepCopy

In that sense I'm not too sure if it makes sense (I could be wrong) to give to option to skipCT via a constructor. This is moreso just a way to track which parser is a deepCopy and which ins't especially when we're calling `parser.Next()`

Reply:
I'm failing to understand why this isn't needed as well 😬. From our discussions, I understood that we don't want to add _created lines as a separate new metric. This means that the original parser needs to skip _created lines when calling `Next()`, but the ephemeral parser in `CreatedTimestamp()` needs to return it

Reply:
update to this can be found in this comment: https://github.com/prometheus/prometheus/pull/14356#discussion_r1682815624

Reply:
>  I'm failing to understand why this isn't needed as well 😬 From our discussions, I understood that we don't want to add _created lines as a separate new metric. 

Not sure what @ArthurSens mean by "this" now. I only refer to the logic how this field is set. There is quite large and complex explanation in commentry how this is set. I feel it would be easier if we kill this and simply allow constructor to set it. In fact `textparse.WithSkipCT` was already implemented for other reasons, so we have it. My proposal is to:

* keep some default (skip = false?)
* ensure that option as we have now e.g `textparse.WithOMParserCTSeriesSkipped()`
* don't change this variable anywhere else (ofc pass true to deep copied for CT logic)

I think this is already implemented by @Maniktherana by a quick look, just I added suggestions below for default and option name 🤗 Thanks!